### PR TITLE
Automatic deployment with Docker

### PR DESCRIPTION
**Pull request by RoyTakanen**
Github action which automatically deploys the code to configured server.

Workflow:

Docker image is automatically made and pushed to (ghcr.io)[ghcr.io] when someone pushes to `main`-branch. After the build this action connects to the server with ssh and pulls the new image and redeploys the container.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+  deploy:
+    needs: [build-and-push-image]
+    name: Deploy to server
+    runs-on: ubuntu-latest
+    steps:
+    - name: Deploy to server
+      uses: fifsky/ssh-action@master
```
Comment (Esinko) [inline .github/workflows/deploy.yml:52]: We should use the more popular `https://github.com/appleboy/ssh-action` for the sake of security. This action is also proven work well in our production environment.
The deploy action will require other minor modifications to suite this change.

**Review by Esinko — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (RoyTakanen) [reply .github/workflows/deploy.yml:52]: I actually tried to use that but I ran into problems with authentication while testing with my own server (and the `fifsky/ssh-action` worked). If the action though works in the production I do not see any problems with that.

Comment (Esinko) [reply .github/workflows/deploy.yml]: Yup! This GH-action is currently in use for at least [testausserveri/testausapis](https://github.com/Testausserveri/testausapis/) and has proven to work well. You may reffer to it's [deploy.yml ](https://github.com/Testausserveri/testausapis/blob/main/.github/workflows/deploy.yml)

**Review by ahnl — APPROVED** [review]

**Comment by ahnl** [comment]
Thank you for your useful contribution! I'll merge this now and configure it asap.

**Pull request merged**